db/comdb2.c

```diff
@@ -8852,6 +8852,38 @@ int check_current_schemas(void)
     return 0;
 }
 
+/**
+ * -- is_valid_dbname
+ * decides whether a given `dbname` is valid database name.
+ */
+int is_valid_dbname(const char *dbname) {
+	// cannot be null.
+	if(dbname == NULL) {
+		logmsg(LOGMSG_ERROR, "DATABASE NAME CANNOT BE EMPTY\n");
+		return -1;
+
+	}
+
+	// cannot be larger than `MAX_DBNAME_LENGTH`
+	size_t len;
+	len = strlen(dbname);
+	if (len >= MAX_DBNAME_LENGTH) {
+		logmsg(LOGMSG_ERROR, "INVALID DATABASE NAME, MUST BE < %d CHARACTERS\n", MAX_DBNAME_LENGTH);
+		return -1;
+	}
+	
+	// alphanumeric and _ are allowed.
+	const char *p = dbname;
+	while (*p++) {
+		if(!isalnum(*p) && *p != '_') {
```

Comment:
I'm not 100% sure what impact remote databases will have.  `_` has special meaning there. Everything else that's a valid SQL identifier should be ok (and things that aren't may be quotable).  Will check regarding maximum supportable length.